src/v/kafka/server/group_manager.cc

```diff
@@ -86,6 +101,57 @@ void group_manager::attach_partition(ss::lw_shared_ptr<cluster::partition> p) {
     _partitions.rehash(0);
 }
 
+ss::future<> group_manager::cleanup_removed_topic_partitions(
+  const std::vector<model::topic_partition>& tps) {
+    using group_info = absl::node_hash_map<group_id, group_ptr>::value_type;
+    return ss::do_for_each(_groups, [this, &tps](group_info& gi) {
+        return gi.second->remove_topic_partitions(tps).then(
+          [this, g = gi.second] {
+              if (!g->in_state(group_state::dead)) {
+                  return ss::now();
+              }
+              auto it = _groups.find(g->id());
+              if (it == _groups.end()) {
+                  return ss::now();
+              }
+              // ensure the group didn't change
+              if (it->second != g) {
+                  return ss::now();
+              }
+              vlog(klog.trace, "Removed group {}", g);
+              _groups.erase(it);
+              _groups.rehash(0);
```

Comment:
nice.

Reply:
i wonder if we can get absl to do this for us automatically if we set capacity to 100% - (1 * entry)